src/rtc.rs

```diff
@@ -276,24 +280,23 @@ impl<'a, I: Instance> Rtc<'a, I> {
         rtc.ier().modify(|_, w| w.taie().clear_bit());
     }
 
-    pub fn is_alarm_triggered(&self) -> bool {
-        ALARM_TRIGGERED.load(Ordering::Relaxed)
-    }
-}
-
-pub fn on_interrupt() {
-    let rtc = unsafe { &*pac::Rtc0::ptr() };
-    // Check if this is actually a time alarm interrupt
-    let sr = rtc.sr().read();
-    if sr.taf().bit_is_set() {
-        rtc.ier().modify(|_, w| w.taie().clear_bit());
-        ALARM_TRIGGERED.store(true, Ordering::SeqCst);
-    }
+    pub async fn wait_for_alarm(&mut self) {
+        WAKER.wait().await;
```

Comment:
you need to subscribe first:

```suggestion
        let wait = WAKER.rubscribe().await;
        wait.wait().await; // you should return a result to propagate errors
```

also, don't you need to unmask interrupt events?

Reply:
Interrupt flag is cleared in the on_interrupt() function and the interrupt is then disabled in the wait_for_alarm() function. Should I modify it and do both in thw wait_for_alarm() ?

Reply:
Right, I saw that. But will a subsequent `wait_for_alarm` call still be able to trigger interrupts? It wasn't clear to me.

Reply:
As long as in the application the user follows the same set_alarm -> set_interrupt then it should work. Will have a try to ensure it is the case

Reply:
Can confirm it works : 

```
loop {
        rtc.wait_for_alarm().await;
        rtc.stop();
        defmt::info!("*** ALARM TRIGGERED! ***");
        let mut alarm1 = rtc.get_datetime();
        alarm1.second += 5;
        rtc.set_alarm(alarm1);
        rtc.set_interrupt(RtcInterruptEnable::RTC_ALARM_INTERRUPT_ENABLE);
        rtc.start();
        rtc.wait_for_alarm().await;
        defmt::info!("new alarm triggered");
        break;
    }
```
Triggers the first alarm after 10 sec then the second one after 5 sec and exits the loop.

Reply:
Oh, I see now. We should probably do away with `set_interrupt` as it creates the possibility of waiting forever if the user forgets to call it or passes an inappropriate mask. Ideally, `set_alarm` would unmask the correct interrupt events when it's called.

Reply:
Good idea, will implement it !